test/scheduler.test.ts

```diff
@@ -130,4 +130,237 @@ describe("Schedule Store", () => {
   test("delete non-existing schedule", async () => {
     await expect(store.schedules.delete("non-existing-id")).rejects.toThrow();
   });
+
+  test("Schedule Store: Create schedule", async () => {
+    const scheduleId = "new-schedule";
+    const cronExpression = "* * * * *"; // Every minute
+
+    const createdSchedule = await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "New Schedule",
+      cronExpression,
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    expect(createdSchedule.id).toBe(scheduleId);
+    expect(createdSchedule.cron).toBe(cronExpression);
+  });
+
+  // this test needs to be discussed
+  test("Schedule Store: Create schedule that exists with same idempotency key", async () => {
+    const scheduleId = "existing-schedule";
+    const cronExpression = "* * * * *"; // Every minute
+
+    // Create the initial schedule
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Existing Schedule",
+      cronExpression,
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Attempt to create a schedule with the same idempotency key, should not throw error, rather return the existing schedule
```

Comment:
Nice find! Let's look in to whats going on. Is it the remote store, local store, or both that are not throwing an error?

Reply:
verified, the behaviour is as per expectation. both stores are not throwing error rather returning an existing schedule.